testcases/OsSecureBoot.py

```diff
@@ -0,0 +1,169 @@
+ - assert physical presence
+   - ensure machine is in a non-secure boot state
+
+
+"""
+
+# Location of the oskernels.tar and oskeys.tar test data files
+# These belong somewhere else, possibly in the code tree?
+# using a temporary link to the github page, this should probably be something more reliable
+URL = "https://github.com/erichte-ibm/op-test/raw/erichte-ibm/os-secure-boot-squashed/test_binaries"
+
+class OsSecureBoot(unittest.TestCase):
+    def setUp(self):
+        conf = OpTestConfiguration.conf
+        self.cv_SYSTEM = conf.system()
+        self.cv_BMC = conf.bmc()
+        self.cv_HOST = conf.host()
+        self.cv_IPMI = conf.ipmi()
+        self.OpIU = InstallUtil()
+
+
+    def getTestData(self, data="keys"):
+        con = self.cv_SYSTEM.console
+        self.OpIU.configure_host_ip()
+
+        fil = "os{}.tar".format(data)
+        url = URL + "/" + fil
+
+        con.run_command("wget {0} -O /tmp/{1}".format(url, fil))
+        con.run_command("tar xf /tmp/{} -C /tmp/".format(fil))
+
+
+    def checkKeysEnrolled(self, enrolled=True):
+        self.cv_SYSTEM.goto_state(OpSystemState.PETITBOOT_SHELL)
+        con = self.cv_SYSTEM.console
+
+        assertfunc = self.assertFalse if enrolled else self.assertTrue
+
+        for k in ["PK", "KEK", "db", "dbx"]:
+            # Size should return a nonzero ascii value when enrolled
+            output = con.run_command("cat /sys/firmware/secvar/vars/{}/size".format(k))
+            assertfunc("0" in output)
+
+            # Data should contain something
+            output = con.run_command("cat /sys/firmware/secvar/vars/{}/data | wc -c".format(k))
+            assertfunc("0" in output)
+
+
+    def checkSecureBootEnabled(self, enabled=True, physical=False):
+        self.cv_SYSTEM.goto_state(OpSystemState.PETITBOOT_SHELL)
+        con = self.cv_SYSTEM.console
+
+        con.run_command("test {} -f /sys/firmware/devicetree/base/ibm,secureboot/os-secureboot-enforcing"
+            .format("" if enabled else "!"))
+
+        if physical:
+            con.run_command("test -f /sys/firmware/devicetree/base/ibm,secureboot/physical-presence-asserted")
+            con.run_command("test -f /sys/firmware/devicetree/base/ibm,secureboot/clear-os-keys")
+
+
+    def assertPhysicalPresence(self):
+        self.cv_SYSTEM.goto_state(OpSystemState.OFF)
+
+        self.cv_BMC.image_transfer("test_binaries/physicalPresence.bin")
+        self.cv_BMC.run_command("cp /tmp/physicalPresence.bin /usr/local/share/pnor/ATTR_TMP")
+        self.cv_BMC.run_command("echo '0 0x283a 0x15000000' > /var/lib/obmc/cfam_overrides")
+        self.cv_BMC.run_command("echo '0 0x283F 0x20000000' >> /var/lib/obmc/cfam_overrides")
+
+        self.cv_IPMI.ipmitool.run("raw 0x04 0x30 0xE8 0x00 0x40 0x00 0x00 0x00 0x00 0x00 0x00 0x00")
+        output = self.cv_IPMI.ipmitool.run("raw 0x04 0x2D 0xE8")
+        self.assertTrue("40 40 00 00" in output)
+        
+        self.cv_SYSTEM.sys_power_on()
```

Comment:
this is where we may need a new state for goto_state()...

Reply:
What would the state be? Something that matches "Physical presence window"?

Reply:
This is a tricky one.. because the system doesn't stop at that state, it goes to a power-off state...

perhaps it's overkill but it feels like we now need to duplicate most of our states to account for a 'physical presence asserted' versus 'physical presence deasserted', with transitions between them.. 

Perhaps there's a simpler way we can start though.. this is a single test-case and I don't think we should deface the entire op-test framework for it, if it can be avoided..

Reply:
It occurred to me that this test depends on OpenBMC anyway for the physical presence assertion, and that reboot is the only one that breaks the states. Would it be okay for this one boot in the test to be the exception to using the state machine reboots? Seems a lot simpler than injecting a new state into the machine for a single test.

Reply:
Yes, let's see how it works that way for now. The important thing is the testcase to not blow-up if running against a non-supported target (Supermicro BMC, FSP, HMC or even OpenBMC without the proper support). Speaking of it, is it possible to test this on Qemu?

Reply:
We should just skip in those cases though, right? OS secureboot is only supported on witherspoon right now. I added checks that look for device tree entries that should be present on compatible hardware, and skip the test if they aren't.

As for Qemu, I have no idea. I'll have to look into that as a possibility.

Reply:
yes, skip tests when not supported.